File: pos-frontend-ng/src/app/shared/models/reports/report-sub-category.ts
import {BasicInterface} from "../basic-interface"; 
import {ReportCategory} from "./report-category";
import {ReportConfig} from "./report-config";

export interface ReportSubCategory extends BasicInterface{
    name:string; 
    
    reportConfig ?:ReportConfig ; 
    
    reportCategory?:ReportCategory ;
}

File: pos-reports/src/main/java/zw/co/hisolutions/pos/reports/entity/ReportSubCategory.java
package zw.co.hisolutions.pos.reports.entity;

import com.fasterxml.jackson.annotation.JsonProperty;
import java.util.List;
import javax.persistence.CascadeType;
import javax.persistence.Entity;
import javax.persistence.ManyToOne;
import javax.persistence.OneToMany;
import javax.persistence.OrderColumn;
import lombok.Data;
import zw.co.hisolutions.pos.common.entity.BaseEntity;

/**
 *
 * @author dgumbo
 */
@Entity
@Data
class ReportSubCategory extends BaseEntity {
    
    @ManyToOne 
    private ReportCategory reportCategory;
     
    private String name;  
    
    @JsonProperty(access = JsonProperty.Access.WRITE_ONLY) 
    @OneToMany(targetEntity=ReportConfig.class, cascade=CascadeType.ALL )
    @OrderColumn
    private List<ReportConfig> reportConfigList;
}
